linetools/isgm/abscomponent.py

```diff
@@ -238,11 +241,11 @@ def add_absline(self, absline, tol=0.1*u.arcsec, chk_vel=True,
         testE = bool(self.Ej == absline.data['Ej'])
         # Now redshift/velocity
         if chk_vel:
-            dz_toler = (1+self.zcomp)*vtoler/3e5  # Avoid Quantity for speed
-            zlim_line = (1+absline.attrib['z'])*absline.analy['vlim']/const.c.to('km/s')
-            zlim_comp = (1+self.zcomp)*self.vlim/const.c.to('km/s')
-            testv = (zlim_line[0] >= (zlim_comp[0]-dz_toler)) & (
-                zlim_line[1] <= (zlim_comp[1]+dz_toler))
+            dz_toler = (1 + self.zcomp) * np.fabs(vtoler) / c_kms  # Avoid Quantity for speed
```

Comment:
It shouldn't be negative, but the user may enter a negative value... maybe safer to generate an error if the vtoler is negative?